### PR TITLE
[HTTP V2] Update V2 Infer API, Fix response error json

### PR DESCRIPTION
**Pull request by CoderHam**
Fix Infer - Update to use new V2 Infer API
Ensure error message responses are in json form

### CODE REVIEW

File: src/servers/http_server_v2.cc [hunk 1]
```diff
@@ -768,18 +768,22 @@ HTTPAPIServerV2::HandleServerReady(
...
-      server_id_);
-
-  evhtp_headers_add_header(
-      req->headers_out,
-      evhtp_header_new(
-          kStatusHTTPHeader, request_status.ShortDebugString().c_str(), 1, 1));
-
-  evhtp_send_reply(
-      req, (ready && (err == nullptr)) ? EVHTP_RES_OK : EVHTP_RES_BADREQ);
+  if (err == nullptr) {
```
Comment (deadeyegoodwin) [inline src/servers/http_server_v2.cc]: Why is this function called "Ready" when it is both live and ready. Why is the model version called "Health" when it is just ready. Please take some time to think of those coming after you into this code and make it as easy as possible for them to understand what is going on.

File: src/servers/http_server_v2.cc [hunk 2]
```diff
@@ -768,18 +768,22 @@ HTTPAPIServerV2::HandleServerReady(
...
+    evhtp_send_reply(req, ready ? EVHTP_RES_OK : EVHTP_RES_BADREQ);
+  } else {
+    rapidjson::Document document;
```
Comment (deadeyegoodwin) [inline src/servers/http_server_v2.cc]: Look carefully at the design doc, live and ready endpoints do not return a json response.

File: src/servers/http_server_v2.cc [hunk 3]
```diff
@@ -998,27 +1010,21 @@ HTTPAPIServerV2::HandleModelMetadata(
 
     evbuffer_add(
         req->buffer_out, model_metadata.c_str(), model_metadata.size());
+    evhtp_send_reply(req, EVHTP_RES_OK);
+  } else {
+    document.Clear();
```
Comment (deadeyegoodwin) [inline src/servers/http_server_v2.cc]: Should make a common routine for this, don't copy-paste. Also this seems unnecessarily complex for such a simple json response. Just create it as a string... will be better performance also.

File: src/servers/http_server_v2.cc (hunk 1 above)
Comment (CoderHam) [reply src/servers/http_server_v2.cc]: Sorry. I will make the necessary changes

File: src/servers/http_server_v2.cc (hunk 2 above)
Comment (CoderHam) [reply src/servers/http_server_v2.cc]: Done

File: src/servers/http_server_v2.cc [hunk 4]
```diff
@@ -271,12 +271,13 @@ class HTTPAPIServerV2 : public HTTPServerV2Impl {
       }
     }
 
+    void SetRequestId(uint64_t request_id) { request_id_ = request_id; }
```
Comment (deadeyegoodwin) [inline src/servers/http_server_v2.cc]: The id is available in the InferenceRequest object, does it need to be maintained here as well?

**Comment by CoderHam** [comment]
Resolved and started one now

> I don't see a CI for this.

**Comment by deadeyegoodwin** [comment]
@CoderHam needs manual rebase

**Comment by CoderHam** [comment]
Changes already picked up the Binary Data PR.

**Pull request closed without merging**